src/db_manager.cpp

```diff
@@ -325,6 +325,47 @@ QStringList DBManager::getUsers()
     return users;
 }
 
+QList<DBManager::BanInfo> DBManager::getBanInfo(QString lookup_type, QString id)
+{
+    QList<BanInfo> return_list;
+    QSqlQuery query;
+    QList<BanInfo> invalid;
+    if (lookup_type == "banid") {
+        query.prepare("SELECT * FROM BANS WHERE ID = ?");
```

Comment:
the only bit that has to change here is the statement itself, the rest doesn't need to be copy-pasted. the last three query.whatever calls can be outside of the if